matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/prune/RedactionEventProcessor.kt

```diff
@@ -104,6 +109,39 @@ internal class RedactionEventProcessor @Inject constructor() : EventInsertLivePr
         }
     }
 
+    /**
+     * Invalidates the number of threads in the main timeline thread summary,
+     * with respect to redactions.
+     */
+    private fun handleTimelineThreadSummaryIfNeeded(
+            realm: Realm,
+            eventToPrune: EventEntity,
+            isLocalEcho: Boolean,
+    ) {
+        if (eventToPrune.isThread() && !isLocalEcho) {
+            val roomId = eventToPrune.roomId
+            val rootThreadEvent = eventToPrune.findRootThreadEvent() ?: return
+            val rootThreadEventId = eventToPrune.rootThreadEventId ?: return
+
+            val numberOfThreads = countThreadReplies(
+                    realm = realm,
+                    roomId = roomId,
+                    rootThreadEventId = rootThreadEventId
+            ) ?: return
+
+            rootThreadEvent.numberOfThreads = numberOfThreads
```

Comment:
`rootThreadEvent.numberOfThreads` I'm assuming this is also number of replies in the thread?

Reply:
Yes, I didn't name it thread replies, while someone may be confused that are the replies (long press reply) within threads. I decided to call a `thread` every individual thread message, but now I don't really like it. A better name might be `threadNormalReplies`, `threadMessages`, `inThreadMessages`